client/src/ui/components/Navigation.tsx

```diff
@@ -1,19 +1,35 @@
+import { useState, useEffect } from "react";
 import { useNavigate } from "react-router-dom";
 import "../assets/styles/Navigation.css";
 import logo from "../assets/images/logo.png";
 import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
-import { faMagnifyingGlass } from "@fortawesome/free-solid-svg-icons";
-import { faUserGroup } from "@fortawesome/free-solid-svg-icons";
-import { faCirclePlay } from "@fortawesome/free-solid-svg-icons";
-import { faUser } from "@fortawesome/free-solid-svg-icons";
-import { faRightFromBracket } from "@fortawesome/free-solid-svg-icons";
-import { faGear } from "@fortawesome/free-solid-svg-icons";
+import {
+  faMagnifyingGlass,
+  faUserGroup,
+  faCirclePlay,
+  faUser,
+  faRightFromBracket,
+  faGear,
+  faBars,
+} from "@fortawesome/free-solid-svg-icons";
 
 const Navigation = () => {
   const navigate = useNavigate();
-  // Finds the current path, make button component, render depending on page
-  // Do this in btn component, pass in "Home" as props, if home == pathname /home, render
-  // Otherwise render default unclicked button
+  const [isMenuOpen, setIsMenuOpen] = useState(false);
+  const [isMobile, setIsMobile] = useState(window.innerWidth <= 650);
+
+  useEffect(() => {
```

Comment:
this logic for resizing is being done in mutliple files, could you make a common function (like in `src/ui/utils` or `/src/ui/hooks/` and use it in the files instead? the function inside `useEffect` can be moved to this common file - you could even make the function take in an input for the innerWidth (so here you'd pass in 650)